### ISSUE TITLE
Custom SFTP server implemented; problem with SSH client from MoveIT Central 

### ISSUE THREAD

**Issue by developer94404** [mscdex/ssh2#468]
The client software:  https://www.ipswitch.com/secure-information-and-file-transfer/moveit-automation

Hello,

I am wondering if I may ask your kind advice or suggestions. First, thanks for your excellent library.

I have implemented a SFTP server using your library, by filling out the necessary code and response to the particular events I need to support. I don't implement these events: MKDIR, RMDIR, REMOVE, SETSTAT, SYMLINK, RENAME since I do not need their functionality.

I have no problems with the WinSCP and FileZilla clients connecting via SFTP to my implemented SFTP server. However, with the client software linked at top (MoveIT Central Automation SSH-SFTP client), I am having problems connecting to the SFTP server.

Their client seems to generate a failure after issuing an OPENDIR and READDIR request to my SFTP server, and disconnects automatically, but no errors occur on the SFTP server end. I don't see any requests to any SFTP events that I don't support, so that doesn't appear to be an issue. WinSCP and FileZilla clients send the same sequence and do not experience errors and correctly connect and list the directory.

Would you know of any potential reason why this client cannot connect and appears to have a general failure "listing the directory" and/or with some given "parameters" in the SFTP events?

Thanks so much!

**Comment by mscdex** [comment]
Can you share the sftp server code and the debug output (at least server side while the client is connected, any client side debug output if available would be nice too)?

**Comment by mscdex** [comment]
@developer94404 Are you still encountering this issue? If so, can you please provide the previously requested information?

**Comment by developer94404** [comment]
Hi mscdex -- my thanks for your replies and your excellent project. 

Apologies for the delayed response - things got busy and fell off the radar.

I am closing this case.  I have fixed this and this was issue was due to my *not* implementing a SFTP event that this particular client expected, to work.  All other widely available clients such as WinSCP and FileZilla, are able to connect without this event being implemented.
